### DISCUSSION TITLE
Hard coded colors

### DISCUSSION THREAD

**Discussion by BananaSupreme** [dotnet/Microsoft.Maui.Graphics.Controls#34] [Ideas]
Hi, I love the goal of the library and would love to contribute! 
I am reading through code to understand what it's doing, and the colours are hard coded, is this the best idea? 
Or am I missing something?

**Comment by jsuarezruiz** [comment]
Hi!. Any contribution will be more than welcome. No, hard coded colors are not a good idea and should not happen. All the controls must use the custom colors defined by the user or fallback in default colors defined in color schemas like https://github.com/dotnet/Microsoft.Maui.Graphics.Controls/blob/main/src/GraphicsControls/Platform/Fluent.cs#L9

**Reply by BananaSupreme** [reply]
Ok, but for example here
 ```
if (checkBox.IsChecked)
            {
                if (checkBox.IsEnabled)
                    canvas.FillColor = Material.Color.Blue.ToColor();
                else
                    canvas.FillColor = Material.Color.Gray3.ToColor();

                canvas.FillRoundedRectangle(x, y, size, size, 2);
            }
```
or here 
```
canvas.FillColor = 
slider.MinimumTrackColor.WithDefault(slider.IsEnabled 
? DefaultCupertinoSliderTrackProgressColor 
: Cupertino.Color.SystemGray.Light.InactiveGray);
```
where 
`const string DefaultCupertinoSliderTrackProgressColor = "#007AFF";`

They are all basically hard coded, what I'm asking is shouldn't we allow the user to edit those schemes? otherwise unless you are using the purest implementation of those patterns the entire library becomes useless... 

EDIT: Okay I just realised the second one allows customizing colors with the WithDefault method, first example exists plenty enough, and secondly changing the base schemes could be a cleaner way then having many style sheets defining the colors for all your different views...
